component/src/main/java/io/siddhi/extension/io/file/function/FileMoveExtension.java

```diff
@@ -0,0 +1,186 @@
+                                "insert into  ResultStream;",
+                        description = "This query moved a particular file to a given path. " +
+                                "The successfulness of the process will be returned as an boolean to the" +
+                                "stream named 'RecordStream'."
+                )
+        }
+)
+public class FileMoveExtension extends FunctionExecutor {
+    private static final Logger log = Logger.getLogger(FileMoveExtension.class);
+    private Attribute.Type returnType = BOOL;
+    @Override
+    protected StateFactory init(ExpressionExecutor[] attributeExpressionExecutors, ConfigReader configReader,
+                                SiddhiQueryContext siddhiQueryContext) {
+        int executorsCount = attributeExpressionExecutors.length;
+        if (executorsCount != 2) {
+            throw new SiddhiAppValidationException("Invalid no of arguments passed to file:move() function, "
+                    + "required 2, but found " + executorsCount);
+        }
+        ExpressionExecutor executor1 = attributeExpressionExecutors[0];
+        ExpressionExecutor executor2 = attributeExpressionExecutors[1];
+        if (executor1.getReturnType() != STRING) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for the first argument of "
+                    + "file:move() function, required " + STRING.toString() + ", but found "
+                    + executor1.getReturnType().toString());
+
+        }
+        if (executor2.getReturnType() != STRING) {
+            throw new SiddhiAppValidationException("Invalid parameter type found for the second argument of "
+                    + "file:move() function,required " + STRING.toString() + ", but found "
+                    + executor2.getReturnType().toString());
+        }
+        return null;
+    }
+
+    @Override
+    protected Object execute(Object[] data, State state) {
+        VFSClientConnector vfsClientConnector = new VFSClientConnector();
+        VFSClientConnectorCallback vfsClientConnectorCallback = new VFSClientConnectorCallback();
+        String sourceFileUri = (String) data[0];
+        String destinationDirUri = (String) data[1];
+        BinaryCarbonMessage carbonMessage = new BinaryCarbonMessage(ByteBuffer.wrap(
+                sourceFileUri.getBytes(StandardCharsets.UTF_8)), true);
+        Map<String, String> properties = new HashMap<>();
+        properties.put(Constants.ACTION, Constants.MOVE);
+        properties.put(Constants.URI, sourceFileUri);
+        String destination = constructPath(destinationDirUri,
+                getFileName(sourceFileUri, validateURLAndGetProtocol(sourceFileUri)));
+        properties.put(Constants.DESTINATION, destination);
+        try {
+            vfsClientConnector.send(carbonMessage, vfsClientConnectorCallback, properties);
+            vfsClientConnectorCallback.waitTillDone(5000, sourceFileUri);
```

Comment:
Please replace `5000` with `WAIT_TILL_DONE` constant.

Reply:
this is fixed.